network/manager.go

```diff
@@ -216,7 +216,14 @@ func (mgr *Manager) StartConnector() {
 // StartListener will start a listener on a given port.
 func (mgr *Manager) StartListener(stop <-chan struct{}) {
 	mgr.wg.Add(1)
-	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr, stop)
+	go handleListening(mgr.log, mgr.wg, mgr.cfg, mgr,
+		func(addr *net.TCPAddr) (Listener, error) {
+			listener, err := net.ListenTCP("tcp", addr)
+			if err != nil {
+				return nil, err
+			}
+			return &TcpListener{listener: listener}, nil
```

Comment:
You can just return the `net.TCPListener` directly here, no need to wrap it.

Reply:
Actually it is necessary, I changed the interface for SetDeadline(time.Duration).

Now, it does the following:

`tcpListener.listener.SetDeadline(time.Now().Add(deadline))`. As I needed to extract interface, I used it also to avoid dealing with time.Now()

Reply:
Why would you need to change it from duration to time? You can just use `time.Now()` in the tests?

Reply:
Obsolete, tests don't rely on it :)